packages/renderer/src/lib/node/NodeDetails.svelte

```diff
@@ -18,14 +17,17 @@ import { NodeUtils } from './node-utils';
 import NodeDetailsSummary from './NodeDetailsSummary.svelte';
 import type { NodeUI } from './NodeUI';
 
-export let name: string;
+interface Props {
+  name: string;
+}
+let { name }: Props = $props();
 
-let node: NodeUI;
-let detailsPage: DetailsPage;
-let kubeNode: V1Node | undefined;
-let kubeError: string;
+let node: NodeUI | undefined = $state(undefined);
+let detailsPage: DetailsPage | undefined = $state(undefined);
+let kubeNode: V1Node | undefined = $state(undefined);
+let kubeError: string | undefined = $state(undefined);
 
-let events: EventUI[] = [];
+let events = $derived($kubernetesCurrentContextEvents.filter(ev => ev.involvedObject.uid === node?.uid));
```

Comment:
```suggestion
let events: EventUI[] = $derived($kubernetesCurrentContextEvents.filter(ev => ev.involvedObject.uid === node?.uid));
```

suggestion (non-blocking): since we have the types above, might as well adding it too